core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -453,40 +452,34 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
   }
 
   /**
-   * Helper method to update quota types counts and quotaTypesEnabled flag.
-   * @param quotaTypeKey The QuotaTypes constant (e.g., QuotaTypes.UserClientIdQuotaEnabled)
-   * @param increment True to increment count, false to decrement
+   * Helper method to update quotaTypesEnabled which is a bitwise OR combination of the enabled quota types.
+   * For example:
+   *  - If UserQuotaEnabled = 2 and ClientIdQuotaEnabled = 1, then quotaTypesEnabled = 3 (2 | 1 = 3)
+   *  - If UserClientIdQuotaEnabled = 4 and UserQuotaEnabled = 1, then quotaTypesEnabled =  (4 | 1 = 5)
+   *  - If UserClientIdQuotaEnabled = 4 and ClientIdQuotaEnabled = 2, then quotaTypesEnabled = 6 (4 | 2 = 6)
+   *  - If all three are enabled (1 | 2 | 4), then quotaTypesEnabled = 7
    */
-  private def updateQuotaTypes(quotaTypeKey: Int, increment: Boolean): Unit = {
-    if (quotaTypeKey == QuotaTypes.NoQuotas) {
-      return
-    }
-    val previousQuotaTypesEnabled = quotaTypesEnabled
-
-    // Update activeQuotaTypes counts
-    activeQuotaTypes.compute(quotaTypeKey, (_, count) =>
-      if (increment) Option(count).getOrElse(0) + 1
-      else if (Option(count).exists(_ > 1)) count - 1
-      else 0
-    )
+  private def updateQuotaTypes(): Unit = {
+    quotaTypesEnabled =  if (clientQuotaCallbackPlugin.isDefined) {
+        QuotaTypes.CustomQuotas
```

Comment:
interesting, so we can have custom quota as well as any combination of user/userclient/client enabled at the same time? (what does a customquota even look like?) let's also add a test for this case (custom can be combined with another quota type)

Reply:
thanks for the details and for the additional test!